### DISCUSSION TITLE
setup failed error in $ forge test --fork-url $SEPOLIA_RPC_URL in lesson 9

### DISCUSSION THREAD

**Discussion by Aishwaria8786** [Cyfrin/foundry-full-course-cu#217] [Q&A] [answered]
I am facing this problem in lesson 9
```
aish123@DESKTOP-8AM50BD:~/foundry-f23/foundry-smart-contract-f23$ forge test --fork-url $SEPOLIA_RPC_URL
[⠰] Compiling...
No files changed, compilation skipped
2023-07-02T16:12:39.282656Z ERROR forge::runner: setUp failed reason="Custom Error d8a3fb52:(0xBf4485c14a627E2eEBA86870A72e60692fA1d56c)" contract=0x34a1d3fff3958843c43ad80f30b94c510645c316

Running 1 test for test/unit/RaffleTest.t.sol:RaffleTest
[FAIL. Reason: Setup failed: Custom Error d8a3fb52:(0xBf4485c14a627E2eEBA86870A72e60692fA1d56c)] setUp() (gas: 0)
Test result: FAILED. 0 passed; 1 failed; finished in 22.61s

Failing tests:
Encountered 1 failing test in test/unit/RaffleTest.t.sol:RaffleTest
[FAIL. Reason: Setup failed: Custom Error d8a3fb52:(0xBf4485c14a627E2eEBA86870A72e60692fA1d56c)] setUp() (gas: 0)

Encountered a total of 1 failing tests, 0 tests succeeded
```
forge test works fine.How to solve this?

**Comment by ghost** [comment]
Hello @Aishwaria8786, only with the error message is going to be difficult to help you, can you format your question according to these [recommendations](https://github.com/Cyfrin/foundry-full-course-f23/blob/main/how-to-ask-a-question.md)?

**Reply by Aishwaria8786** [reply]
I am in lesson 9.I run forge test and all test passed. But when I am running forge test --fork-url $SEPOLIA_RPC_URL ,it is showing this particular problem.

**Reply by ghost** [reply]
Verify if the variables that you're using are avaliable on the testnet. Also verify If you're doing a call to `fulfillRandomWords` or something similar that you're not allowed to do on a testnet.

**Comment by shikhar229169** [comment]
Hey, @Aishwaria8786, please provide the code in which you are facing problem.

**Reply by Aishwaria8786** [reply]
I am in lesson 9.I run forge test and all test passed. But when I am running forge test  --fork-url $SEPOLIA_RPC_URL ,it is showing this particular problem.

**Comment by shedrick4570** [comment]
Without seeing any code, code this also be a set up issue with Chainlink or the private key?

I personally had this same error, but I was only getting it while testing locally (forked was fine). I missed adding deployerKey to the createSubscription functions.

**Comment by Aishwaria8786** [comment]
I surprisingly solved the issue by by changing the subscriptionid to 0 in helperConfig.s.sol file . I would like to advise all to watch Patrick sir's video slowly.

**Reply by ghost** [reply]
Don't forget to mark this answer as the solution,

**Comment by Aishwaria8786** [comment, marked as answer]
I found the actual solution which is I create subscription on chainlink from one account but in env file I put private key of another account.I corrected it now it works fine with subscriptionid .
